force-app/main/default/classes/RD2_ETableController.cls

```diff
@@ -365,4 +365,35 @@ public with sharing class RD2_ETableController {
         
         return permissions.canRead(npe03__Recurring_Donation__c.SObjectType, fields);
     }
+
+    private without sharing class RDScheduleClass {
+        public Map<Id, List<RecurringDonationSchedule__c>> queryRows(Set<Id> recordIds) {
+            Map<Id, List<RecurringDonationSchedule__c>> mapRDSchedule = new Map<Id, List<RecurringDonationSchedule__c>>();
+            for(RecurringDonationSchedule__c schedule : [   SELECT Active__c, 
+                                                                Campaign__c, 
+                                                                Campaign__r.Name, 
+                                                                StartDate__c, 
+                                                                EndDate__c, 
+                                                                DayOfMonth__c, 
+                                                                InstallmentAmount__c, 
+                                                                InstallmentFrequency__c, 
+                                                                InstallmentPeriod__c, 
+                                                                PaymentMethod__c, 
+                                                                RecurringDonation__c, 
+                                                                IsPause__c, 
+                                                                StatusReason__c 
+                                                            FROM RecurringDonationSchedule__c
+                                                            WHERE RecurringDonation__c IN :recordIds
+                                                                AND RecurringDonation__r.npe03__Contact__c = :USER_UserService.getContactIdByUserId()
```

Comment:
Why have a filter on both RecurringDonation and Contact? Isn't one sufficient - especially if you've already narrowed down the specific RD Id's you want the schedule records for?

Reply:
I agree, I updated it.